tools/tpm2_hash.c

```diff
@@ -57,6 +57,11 @@ struct tpm_hash_ctx {
     TSS2_SYS_CONTEXT *sapi_context;
 };
 
+static tpm_hash_ctx ctx = {
+    .hierarchyValue = TPM_RH_NULL,
+    .halg = TPM_ALG_SHA1,
+};
```

Comment:
You didn't answer my previous comment about passing a void * to the callbacks instead of using global variables.

Reply:
We could definitely do that, but they would need to malloc there structure in the onstart() routine, wasn't sure if the complexity was worth it, given that everything is single threaded.

Reply:
Right, let's just keep it as global vars for now to keep the patch simpler and we can later explore if the added complexity is worth it.

Reply:
If you really want to put the tool context structures on the heap, I can do it, I don't feel *that* strongly about it. I have the rest of your changes queued up locally, we can revisit tomorrow. Thanks.

Reply:
No strong opinion really, I trust your judgment since you did the change and have a better mental model of the code.

Reply:
@williamcroberts thanks a lot for addressing all the things I pointed out.

I'll work on porting the tools tomorrow based on the branch you left today.

Reply:
@martinezjavier ill upload the changes to your comments and merge into this branch, feel free to submit PRs for this to that branch.

Reply:
@williamcroberts great, I'll work on top of that branch then and submit PR so you can squash them before pushing to master.